platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -383,6 +386,15 @@ private void logout() {
 
         SecurityTokenHolder tokenHolder = ((SecurityTokenHolder) session.getAttribute(
                 SecurityConstants.SAML_ASSERTION));
+
+        org.apache.shiro.subject.Subject subject = ThreadContext.getSubject();
+        Arrays.stream(System.getProperty("security.audit.roles")
+                .split(","))
+                .filter(subject::hasRole)
+                .findFirst()
+                .ifPresent(p -> SecurityLogger.audit("Subject with admin privileges has logged out",
```

Comment:
I saw this in another PR as well...I'm not fond of the use of `ifPresent` solely for side-effects. If you're not using the result from the `Optional` above, you should be calling the `isPresent` terminal op instead and getting a boolean result.

```java
boolean hasRole = ...
    .isPresent();
if (hasRole) {
    SecurityLogger.audit(...);
}
```